cli/compose/loader/loader.go

```diff
@@ -427,9 +427,21 @@ func loadVolumes(source types.Dict) (map[string]types.VolumeConfig, error) {
 		return volumes, err
 	}
 	for name, volume := range volumes {
-		if volume.External.External && volume.External.Name == "" {
-			volume.External.Name = name
-			volumes[name] = volume
+		if volume.External.External {
+			template := "both external and %s are specified for volume %q. Only one is allowed."
```

Comment:
It reads a little strangely for this to start with a lowercase letter (as errors should) and have a second sentence that starts with uppercase. In general I try to avoid multiple sentences in an error string. How about `conflicting parameters "external" and %q specified for volume %q`?